### PR TITLE
implemented admin lobby

### PR DESCRIPTION
**Pull request by zyfers**

### CODE REVIEW

File: server/src/main/java/com/strategists/game/controller/PlayerController.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public List<Player> getPlayers() {
 	@PostMapping
 	public Player addPlayer(@RequestBody Map<String, Object> body) {
 		Assert.state(gameService.isLobbyState(), "Can't add players to active game!");
-		return playerService.addPlayer((String) body.get(Keys.USERNAME), (double) body.get(Keys.CASH));
```
Comment (shubham1chawla) [inline server/src/main/java/com/strategists/game/controller/PlayerController.java:47]: Why this change?

File: web/src/App.css [hunk 2]
```diff
@@ -13,3 +13,18 @@ h2 {
   text-align: center;
   line-height: 72px;
 }
+
+.ant-tabs-nav {
+  display: flex;
+}
+.ant-tabs-nav .ant-tabs-tab {
```
Comment (shubham1chawla) [inline web/src/App.css:20]: Add a space here.

Comment (shubham1chawla) [reply web/src/App.css:20]: enter

File: web/src/components/AdminLobby.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+const AdminLobby = () => {
+  const dispatch = useDispatch();
+  const players = useSelector((state: any) => state.lobby.players);
+  const [form] = Form.useForm();
+
+  useEffect(() => {
+    axios.get('/api/players').then(async (res) => {
+      await res.data.forEach((player: any) => {
+        dispatch(addPlayer({ name: player.username, cash: player.cash }));
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:17]: Rename the key `name` in the state to `username` for consistency.

File: web/src/components/AdminLobby.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      });
+    });
+  }, []);
+
+  const deleteFromList = async (item: Player) => {
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:22]: You can rename this method as `kickPlayer`, same as backend, for consistency.

File: web/src/components/AdminLobby.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    console.log(item);
+    await axios
+      .delete('/api/players', {
+        data: {
+          username: item.name,
+        },
+      })
+      .then(() => {
+        dispatch(removePlayer(item));
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:31]: Same here `kickPlayer`

File: web/src/components/AdminLobby.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      });
+  };
+
+  const onFinish = async (values: any) => {
+    console.log('Success:', values);
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:36]: Remove the consoles.

File: web/src/components/AdminLobby.tsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        data: {
+          username: item.name,
+        },
+      })
+      .then(() => {
+        dispatch(removePlayer(item));
+      });
+  };
+
+  const onFinish = async (values: any) => {
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:35]: Probably rename `values` with `player`

File: web/src/components/AdminLobby.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        cash: player.coins,
+      })
+      .then((response) => {
+        dispatch(addPlayer(player));
+        form.resetFields();
+      });
+  };
+
+  const onFinishFailed = (errorInfo: any) => {
+    console.log('Failed:', errorInfo);
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:53]: console.error

File: web/src/components/AdminLobby.tsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+            <List.Item
+              style={list}
+              extra={
+                <CloseCircleOutlined
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    deleteFromList(item);
+                  }}
+                  style={{
+                    fontSize: '20px',
```
Comment (shubham1chawla) [inline web/src/components/AdminLobby.tsx:123]: Can we move this `fontSize` outside, as a constant. Suggestion - make list styling constant and defined in App.css

Comment (shubham1chawla) [reply web/src/components/AdminLobby.tsx:123]: Let's use css variables if not.

File: web/src/redux/admin/lobby/lobbyActions.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import { Player } from './lobbyReducer';
+
+export const addPlayer = (player: Player) => {
+  return {
+    type: ADD_PLAYER,
+    payload: player,
+  };
+};
+
+export const removePlayer = (player: Player) => {
```
Comment (shubham1chawla) [inline web/src/redux/admin/lobby/lobbyActions.ts:12]: `kickPlayer`

File: web/src/redux/admin/lobby/lobbyReducer.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import { ADD_PLAYER } from './lobbyTypes';
+import { REMOVE_PLAYER } from './lobbyTypes';
+
+export interface Player {
+  name: string;
```
Comment (shubham1chawla) [inline web/src/redux/admin/lobby/lobbyReducer.ts:5]: `username`

**Review by shubham1chawla — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/strategists/game/controller/PlayerController.java (hunk 1 above)
Comment (zyfers) [reply server/src/main/java/com/strategists/game/controller/PlayerController.java:47]: The object that we were receiving had cash as Integer type and java can't convert Integer to double

File: web/src/components/AdminLobby.tsx (hunk 7 above)
Comment (zyfers) [reply web/src/components/AdminLobby.tsx:35]: i am already making player object from values. Can't have two variables with same name

**Review by shubham1chawla — APPROVED** [review]
